src/frontend.rs

```diff
@@ -71,27 +71,42 @@ impl fmt::Display for Value {
     }
 }
 
+/// SignedPod is a wrapper on top of backend::SignedPod, which additionally stores the
+/// string<-->hash relation of the keys.
 #[derive(Clone, Debug)]
 pub struct SignedPod {
-    pub params: Params,
-    pub id: PodId,
-    pub kvs: HashMap<String, Value>,
+    pub pod: backend::SignedPod,
+    // `string_key_map` is a hashmap to store the relation between key strings and key hashes
+    // TODO review if maybe store it as <Hash, String>, so that when iterating we can get each
+    // hash respective string
+    string_key_map: HashMap<String, crate::Hash>,
```

Comment:
added it as `HashMap<String, Hash>` but with the `TODO` note, since currently we're not using this mapping anywhere so didn't know what would be more useful